ArchiSteamFarm/Steam/Integration/SteamChatMessage.cs

```diff
@@ -0,0 +1,203 @@
+
+					// We can never have more characters than bytes used, so this covers the worst case of 1-byte characters exclusively
+					char[] lineChunk = charPool.Rent(bytesToTake);
+
+					try {
+						// We have to reset the decoder prior to using it, as we must discard any amount of bytes read from previous incomplete character
+						decoder.Reset();
+
+						int charsUsed = decoder.GetChars(lineBytes, lineBytesRead, bytesToTake, lineChunk, 0, false);
+
+						switch (charsUsed) {
+							case <= 0:
+								throw new InvalidOperationException(nameof(charsUsed));
+							case >= 2 when (lineChunk[charsUsed - 1] == '\\') && (lineChunk[charsUsed - 2] != '\\'):
+								// If our message is of max length and ends with a single '\' then we can't split it here, because it escapes the next character
+								// Instead, we'll cut this message one char short and include the rest in the next iteration
+								charsUsed--;
+
+								break;
+						}
+
+						int bytesUsed = Encoding.UTF8.GetByteCount(lineChunk, 0, charsUsed);
+
+						if (lineBytesRead > 0) {
+							messagePartBytes += ContinuationCharacterBytes;
+							messagePart.Append(ContinuationCharacter);
+						}
+
+						lineBytesRead += bytesUsed;
+
+						messagePartBytes += bytesUsed;
+						messagePart.Append(lineChunk, 0, charsUsed);
+					} finally {
+						charPool.Return(lineChunk);
+					}
+
+					if (lineBytesRead < lineBytes.Length) {
+						messagePartBytes += ContinuationCharacterBytes;
+						messagePart.Append(ContinuationCharacter);
+					}
+
+					// Check if we still have room for one more line
+					if (messagePartBytes + NewlineWeight + ReservedEscapeMessageBytes <= maxMessageBytes) {
+						continue;
+					}
+
+					yield return messagePart.ToString();
+
+					messagePartBytes = prefixBytes;
+					messagePart.Clear();
+					messagePart.Append(steamMessagePrefix);
+				}
+			}
+
+			if (messagePart.Length <= prefixLength) {
+				yield break;
+			}
+
+			yield return messagePart.ToString();
+		}
+
+		internal static bool IsValidPrefix(string steamMessagePrefix) {
+			if (string.IsNullOrEmpty(steamMessagePrefix)) {
+				throw new ArgumentNullException(nameof(steamMessagePrefix));
+			}
+
+			return GetPrefixMesageBytes(Escape(steamMessagePrefix)) <= MaxMessagePrefixBytes;
+		}
+
+		internal static string Unescape(string message) {
+			if (string.IsNullOrEmpty(message)) {
+				throw new ArgumentNullException(nameof(message));
+			}
+
+			return message.Replace("\\[", "[", StringComparison.Ordinal).Replace("\\\\", "\\", StringComparison.Ordinal);
+		}
+
+		private static string Escape(string message) {
+			if (string.IsNullOrEmpty(message)) {
+				throw new ArgumentNullException(nameof(message));
+			}
+
+			return message.Replace("\\", "\\\\", StringComparison.Ordinal).Replace("[", "\\[", StringComparison.Ordinal);
+		}
+
+		private static int GetPrefixMesageBytes(string escapedSteamMessagePrefix) {
```

Comment:
In all other cases we had the word "message" followed by the word "prefix" and not the other way round (like here). You also missed one `s` in there 😉

Reply:
🙄